packages/experience-builder-sdk/src/hooks/useFetchExperience.test.ts

```diff
@@ -27,249 +26,223 @@ describe('useFetchExperience', () => {
+    try {
+      await act(() => res.result.current.fetchBySlug({ experienceTypeId, slug, localeCode }));
+    } catch (e) {
+      expect((e as Error).message).toBe(
+        `More than one experience with identifier: ${JSON.stringify({ slug })} was found`
+      );
+    }
 
-    await waitFor(() => expect(result.current.error).toBeUndefined());
+    expect(res.result.current.isFetching).toBe(false);
   });
 
-  it('should return an error if experienceTypeId is not defined', async () => {
-    const initialProps = {
-      client: clientMock,
-      mode: 'preview' as ExternalSDKMode,
-      slug,
-      localeCode,
-      experienceTypeId: undefined,
-    };
-
-    const { result, rerender } = renderHook(useFetchExperience, {
-      // @ts-expect-error undefined is not allowed through types, but it can still happen if invoked from plain js
-      initialProps,
+  it('should throw an error if experienceTypeId is not defined', async () => {
+    const res = renderHook((props) => useFetchExperience(props), {
+      initialProps: { client: clientMock, mode: 'preview' as ExternalSDKMode },
     });
 
-    await waitFor(() =>
-      expect(result.current.error?.message).toBe(
+    try {
+      await act(() =>
+        // @ts-expect-error undefined is not alloed through types, but it can still happen if invoked from a plain js
+        res.result.current.fetchBySlug({ experienceTypeId: undefined, slug, localeCode })
+      );
+    } catch (e) {
+      expect((e as Error).message).toBe(
         'Failed to fetch experience entities. Required "experienceTypeId" parameter was not provided'
-      )
-    );
-
-    rerender({ ...initialProps, experienceTypeId });
+      );
+    }
 
-    await waitFor(() => expect(result.current.error).toBeUndefined());
+    expect(res.result.current.isFetching).toBe(false);
   });
 
-  it('should return an error if slug and id are both undefined, then when one becomes defined the error should be undefined', async () => {
-    const initialProps: Parameters<typeof useFetchExperience>[0] = {
-      client: clientMock,
-      mode: 'preview' as ExternalSDKMode,
-      slug: undefined,
-      id: undefined,
-      experienceTypeId,
-      localeCode,
-    };
-
-    const { result, rerender } = renderHook(useFetchExperience, {
-      initialProps,
+  it('should throw an error if slug is not defined', async () => {
+    const res = renderHook((props) => useFetchExperience(props), {
+      initialProps: { client: clientMock, mode: 'preview' as ExternalSDKMode },
     });
 
-    await waitFor(() => {
-      expect(result.current.error?.message).toBe(
-        'Either slug or id must be provided to useFetchExperience'
+    try {
+      await act(() =>
+        // @ts-expect-error undefined is not alloed through types, but it can still happen if invoked from a plain js
+        res.result.current.fetchBySlug({ experienceTypeId, slug: undefined, localeCode })
       );
+    } catch (e) {
+      expect((e as Error).message).toBe(
```

Comment:
[nitpick]
Same feedback as above that `try/catch` doesn't work in jest tests as intended. Use `toThrow` instead.